src/citrine/resources/project.py

```diff
@@ -365,3 +365,117 @@ def register(self, name: str, description: Optional[str] = None) -> Project:
 
         """
         return super().register(Project(name, description))
+
+
+    def search(self, search_params: Optional[dict] = None, 
+                page: Optional[int] = None, per_page: int = 100) -> Iterable[Project]:
+        """
+        Search for projects matching the desired name, description, and/or status by exact
+        match or substring match, as specified by the search_params argument. Defaults to
+        no search criteria.
+        
+        Like list, this method allows for pagination. This differs from the list function, 
+        because it makes a POST request to resourceType/search with search fields in a post body.
+
+        Leaving page and per_page as default values will yield all elements in the
+        collection, paginating over all available pages.
+
+        Leaving search_params as its default value will return mimic the behavior of 
+        a full list with no search parameters.
+
+        Parameters
+        ---------
+        search_params: dict, optional
+            A dict representing the body of the post request that will be sent to the
+            search endpoint to filter the results ie.
+            {
+                "search_params": {
+                    "name": {
+                        "value": "Polymer",
+                        "search_method": "SUBSTRING"
+                    },
+                    "description": {
+                        "value": "polymer chain length",
+                        "search_method": "SUBSTRING"
+                    },
+                    "status": {
+                        "value": "not-started",
+                        "search_method": "EXACT"
+                    }
+                }
+            }
+            The dict can constain any combination of (one or all) search specifications
+            for the name, description, and status fields of a project. For each parameter
+            specified, the "value" to match, as well as the "search_method" must be provided.
+            The available search_methods are "SUBSTRING" and "EXACT". The example above demonstrates
+            the input necessary to list projects with names including the word "Polymer,"
+            descriptions containing the phrase "polymer chain length," and statuses exactly
+            equal to "not-started."
+        page: int, optional
+            The "page" of results to list. Default is to read all pages and yield
+            all results.  This option is deprecated.
```

Comment:
This is a new function, so we don't have to include `page` and `per_page` if they are deprecated.